grr/server/grr_response_server/gui/ui/components/client_details/entry_history_dialog/entry_history_dialog.ts

```diff
@@ -0,0 +1,54 @@
+import {Component, Inject} from '@angular/core';
+import {MAT_DIALOG_DATA} from '@angular/material/dialog';
+import {Client} from '@app/lib/models/client';
+
+export type EntryType =
+    'primitive'|'timestamp'|'size'|'user-list'|'interface-list'|'volume-list';
+
+export interface EntryHistoryDialogParams {
+  path: string;
+  type: EntryType;
+  clientVersions: Client[];
```

Comment:
In your interface definitions, make all properties immutable if possible: Use `readonly` and `ReadonlyArray`.